### DISCUSSION TITLE
help with authorization

### DISCUSSION THREAD

**Discussion by drabi-he** [nanomq/nanomq#1290] [Q&A]
hello everyone
the problem i have is when the credentials are wrong the client keep trying to reconnect
i want to stop that and send and error to the client informing them the credentials are wrong
any idea how i can accomplish that.
the check for the authorization happens in the broker.c file line : 309 => 313

**Comment by JaylinYu** [comment]
which version are you using? Have you enabled TLS? plz share your config file

**Comment by JaylinYu** [comment]
seems like I misunderstood this question. 
You can accomplish this by using MQTT 5.
